metering/llms.py

```diff
@@ -0,0 +1,241 @@
+import atexit
+import logging
+import os
+from time import time
+from metering.ingest import create_ingest_client
+from metering.ingest.api_client import create_ingest_payload
+
+
+INPUT_TOKENS_METER_API_NAME = "input_tokens"
+OUTPUT_TOKENS_METER_API_NAME = "output_tokens"
+ANTHROPIC_PROVIDER = "anthropic"
+OPENAI_PROVIDER = "openai"
+COHERE_PROVIDER = "cohere"
+VERTEXAI_PROVIDER = "vertexai"
+UNKNOWN_PROVIDER = "unknown"
+
+logger = logging.getLogger(__name__)
+
+
+def customer_id_getter(kwargs):
+    return kwargs.get("customer_id", "test_customer")
+
+
+def aflo_dimensions_getter(kwargs):
+    return kwargs.get("aflo_dimensions", {})
+
+
+def meter_llm(
+    customer_id_getter=customer_id_getter,
+    dimensions_getter=aflo_dimensions_getter,
+    metering_client=None,
+):
+    if metering_client is None:
+        metering_client = create_ingest_client(os.environ.get("API_KEY"))
+        atexit.register(metering_client.shutdown)
+    else:
+        atexit.register(metering_client.shutdown)
```

Comment:
the metering client should be a singleton, so you need to add it to the module scope.

```python
__metering_client = None

if metering_client = None:
    if __metering_client = None:
        __metering_client = ....
        atexit(....)
    metering_client = __metering_client
```

Reply:
or use the global keyword